R/transformAbundance.R

```diff
@@ -0,0 +1,400 @@
+#'
+#'     \item{"log10" }{log10 transform can be used for reducing the skewness of the data.
+#'     ($log_{10}x$, where $x$ is a single value of data.)}
+#'
+#'     \item{"log10p }{log10 transform can be used for reducing the skewness of the data.
+#'     With log10p, pseudo count is added before log10 transformation.
+#'     ($log_{10}x$, where $x$ is a single value of data.)}
+#'
+#'     \item{"hellinger" }{Hellinger transform can be used for reducing the impact of
+#'     extreme data points. ($\sqrt{\frac{x}{x_{tot}}}$, where $x$ is a single value and $x_{tot}$ is the sum of
+#'     all values.)}
+#'
+#'     \item{"identity" }{Identity is the data itself without transform.}
+#'
+#'     \item{"clr" }{Centered log ratio (clr) transform can be used for reducing the
+#'     skewness of data and for centering it.
+#'     ($log_{10}x_{r}) - log_{10}µ_{r}$, where $x_{r}$ is a single relative value, $µ_{r}$ is
+#'     the mean of relative values of whole sample.}
+#'
+#' }
+#'
+#' @references
+#' If there is references Add information here......................................
+#'
+#' @return \code{transformAbundance} returns \code{x} with additional, transformed
+#' abundance table named \code{*name*} in the \code{\link{assay}}.
+#' \code{getTransformAbundance} returns transformed abundance table.
+#'
+#' @seealso
+#' If there is something Add information here......................................
+#'
+#' @name transformAbundance
+#' @export
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' data(esophagus)
+#' x <- esophagus
+#'
+#' # By default, returns object with identity table
+#' x <- transformAbundance(x)
+#' assays(x)$identity
+#'
+#' # By specifying, it is possible to apoly different transformations, e.g. clr transform
+#' # Pseudocount can be added by giving the value TRUE. Then pseudocount is 1.
+#' x <- transformAbundance(x, transform="clr", pseudocount=TRUE)
+#' assays(x)$clr
+#'
+#' # Name of the stored table, can be specified. Also, the target of transform
+#' # can be specified with "abund_values". I addition to pseudocount=TRUE, it is
+#' # also possible to specify it by giving numeric value.
+#' x <- transformAbundance(x, transform="hellinger", name="test", pseudocount=5)
+#' assays(x)$test
+#' x <- transformAbundance(x, transform="Z", abund_values="test")
+#' assays(x)$Z
+#' # Z-transform can also be done for features
+#' x <- transformAbundance(x, transform="Z", pseudocount=TRUE, target="features")
+#' assays(x)$Z
+#'
+#' # With getTransformAbundance, it is possible to get only the transformed abundance
+#' # table without being stored in assays.
+#' getTransformAbundance(x, transform="log10", pseudocount=1)
+#'
+#'
+NULL
+
+#' @rdname transformAbundance
+#' @export
+setGeneric("transformAbundance", signature = c("x"),
+           function(x,
+                    abund_values = "counts",
+                    transform = "identity",
```

Comment:
If you include all available options like `transform = c("clr", "log10",...)` in the generic and the method, you can easily check the validity via `transform  <- match.arg(transform)`

Reply:
Changed